kratos/python_scripts/analysis_stage.py

```diff
@@ -48,11 +48,14 @@ def Run(self):
         self.RunSolutionLoop()
         self.Finalize()
 
+    def KeepAdvancingSolutionLoop(self):
```

Comment:
please add some documentation abt what this function does, since the name is not really self-explaining

Otherwise approving, since this way discussed before :+1:

Reply:
A couple lines of documentation added in the method itself.

Reply:
perfect, thanks